File: PolicyPlus/BitReinterpretation.vb
Imports System.Runtime.InteropServices

<StructLayout(LayoutKind.Explicit)>
Public Structure ReinterpretableDword
    <FieldOffset(0)> Public Signed As Integer
    <FieldOffset(0)> Public Unsigned As UInteger
End Structure

<StructLayout(LayoutKind.Explicit)>
Public Structure ReinterpretableQword
    <FieldOffset(0)> Public Signed As Long
    <FieldOffset(0)> Public Unsigned As ULong
End Structure

File: PolicyPlus/BitReinterpretation.cs
using System.Runtime.InteropServices;

namespace PolicyPlus
{

    [StructLayout(LayoutKind.Explicit)]
    public struct ReinterpretableDword
    {
        [FieldOffset(0)]
        public int Signed;
        [FieldOffset(0)]
        public uint Unsigned;
    }

    [StructLayout(LayoutKind.Explicit)]
    public struct ReinterpretableQword
    {
        [FieldOffset(0)]
        public long Signed;
        [FieldOffset(0)]
        public ulong Unsigned;
    }
}